aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -2,84 +2,164 @@
 
 #include <cmath>
 #include <iostream>
+#include "ATen/CPUApplyUtils.h"
 #include "ATen/Dispatch.h"
 #include "ATen/Parallel.h"
-#include "ATen/cpu/vec256/vec256.h"
 #include "ATen/cpu/vec256/functional.h"
+#include "ATen/cpu/vec256/vec256.h"
 #include "ATen/native/cpu/CapabilityDispatch.h"
 
 namespace at { namespace native {
 namespace {
 
 using namespace vec256;
 
-template <class scalar_t, class F>
-static void parallel_apply(Tensor& result, const Tensor& self, F f) {
-  internal::init_tbb_num_threads();
-
-  static tbb::affinity_partitioner ap;
-
-  auto arr_out = result.data<scalar_t>();
-  auto arr_in = self.data<scalar_t>();
-  int64_t size = self.numel();
-  if (size < internal::TBB_GRAIN_SIZE) {
-    map(f, arr_out, arr_in, size);
-  } else {
-    tbb::parallel_for(
-        tbb::blocked_range<int64_t>(0, size, internal::TBB_GRAIN_SIZE),
-        [&](const tbb::blocked_range<int64_t>& r) {
-          map(f, arr_out + r.begin(), arr_in + r.begin(), r.end() - r.begin());
-        },
-        ap);
-  }
-}
-
 static void abs_kernel(Tensor& result, const Tensor& self) {
   AT_DISPATCH_ALL_TYPES(self.type(), "abs", [&] {
-    parallel_apply<scalar_t>(
+    CPU_tensor_parallel_kernel_apply2<scalar_t, scalar_t>(
```

Comment:
Ok, we should really do something about those names, because they are really long and not very meaningful. I'd start with getting rid of the `_tensor` part, because we don't apply over anything else (and can just overload in any other case). Why not just `CPU_kernel_apply2`? Can you remind me why do we even keep the non-parallel `apply` functions?

Reply:
Functions like "bernoulli" require sequential processing. We can definitely revisit the names.